src/Plugins/DBFTPlugin/Consensus/ConsensusService.OnMessage.cs

```diff
@@ -82,7 +82,8 @@ private void OnPrepareRequestReceived(ExtensiblePayload payload, PrepareRequest
             if (message.Version != context.Block.Version || message.PrevHash != context.Block.PrevHash) return;
             if (message.TransactionHashes.Length > neoSystem.Settings.MaxTransactionsPerBlock) return;
             Log($"{nameof(OnPrepareRequestReceived)}: height={message.BlockIndex} view={message.ViewNumber} index={message.ValidatorIndex} tx={message.TransactionHashes.Length}");
-            if (message.Timestamp <= context.PrevHeader.Timestamp || message.Timestamp > TimeProvider.Current.UtcNow.AddMilliseconds(8 * neoSystem.Settings.MillisecondsPerBlock).ToTimestampMS())
+            TimeSpan blockTime = GetBlockGenTime();
```

Comment:
One thing that should be changed regarding consensus part: it's better to bind `blockTime` to the consensus context, because this value can not be changed within the block processing frame. Check how it implemented in https://github.com/nspcc-dev/dbft/pull/147, in particular, https://github.com/nspcc-dev/dbft/blob/b5dd36eb90f791efdf137069d2a8197145ef92ca/context.go#L250.

So we initialize the context-bound `blockTime` at view 0, and then reuse it from every other handler until next block is accepted. It's needed because these repetitive calls to Policy state (`GetBlockGenTime()`) are useless.